packages/components/src/modal/index.tsx

```diff
@@ -170,13 +170,28 @@ function UnforwardedModal(
 		[ hasScrolledContent ]
 	);
 
-	const onOverlayPress: React.PointerEventHandler< HTMLDivElement > = (
-		event
-	) => {
-		if ( event.target === event.currentTarget ) {
-			event.preventDefault();
-			onRequestClose( event );
-		}
+	let pressTarget: EventTarget | null = null;
+	const overlayPressHandlers: {
+		onPointerDown: React.PointerEventHandler< HTMLDivElement >;
+		onPointerUp: React.PointerEventHandler< HTMLDivElement >;
+	} = {
+		onPointerDown: ( event ) => {
+			if ( event.isPrimary && event.target === event.currentTarget ) {
+				pressTarget = event.target;
+				// Avoids loss of focus yet also leaves `useFocusOutside`
+				// practically useless with its only potential trigger being
+				// programmatic focus movement. TODO opt for either removing
+				// the hook or enhancing it such that this isn't needed.
+				event.preventDefault();
+			}
+		},
+		// Requests closing only for primary pointer/button and if the
+		// target was the same as the one that was pressed.
+		onPointerUp: ( { target, isPrimary, button } ) => {
+			const isSameTarget = target === pressTarget;
+			pressTarget = null;
+			if ( isPrimary && button === 0 && isSameTarget ) onRequestClose();
+		},
```

Comment:
I might be missing something obvious, but what stops us from simply doing this?

```suggestion
		onClick: ( event ) => {
			if ( event.target === event.currentTarget ) {
				onRequestClose();
			}
		},
```

Reply:
It wasn't immediately obvious to me. I tried that first and it works except for supporting the ability to drag back over the modal to cancel the click. When `mousedown` is on the overlay and the `mouseup` is on the modal that's still a `click` from the perspective of the overlay (and `target` is `currentTarget`).

Reply:
Oh I see, good catch. Could we add a code comment about that too? I'm sure others will wonder the same.

Unfortunately this doesn't seem to be accurately testable in Jest, so I'll take note of it as a potential test to add when we set up Playwright tests for components.